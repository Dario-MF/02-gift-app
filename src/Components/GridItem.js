import React from 'react'
import PropTypes from 'prop-types'

const GridItem = ({ url, title }) => {

    return (
        <div className={'card card_item'}>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default GridItem
