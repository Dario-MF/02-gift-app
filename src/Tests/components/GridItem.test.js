import React from 'react'
import { shallow } from "enzyme"
import GridItem from "../../Components/GridItem"

describe('tests for <GridItem />', () => {
    const title = 'titulo'
    const url = 'https//:localhost/image.jpg'
    const wrapped = shallow(<GridItem title={title} url={url} />)

    test('snapshot <GridItem />', () => {


        expect(wrapped).toMatchSnapshot()
    })

    test('comprobar render de p', () => {
        const p = wrapped.find('p')

        expect(p.text().trim()).toBe(title)
    })

    test('comprobar que img recibe props correctamente', () => {
        const img = wrapped.find('img')

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('comprobar className de div', () => {
        const divClassName = wrapped.find('div').prop('className')

        expect(divClassName.includes('card_item')).toBe(true)

    })




})
