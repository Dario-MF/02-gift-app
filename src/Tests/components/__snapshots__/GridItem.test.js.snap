// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <GridItem /> snapshot <GridItem /> 1`] = `
<div
  className="card card_item"
>
  <img
    alt="titulo"
    src="https//:localhost/image.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
